<?xml version="1.0"?>
<!--
  Launch file alternativo que incluye el Action Server en lugar del servicio simple.
  
  Uso:
    ros2 launch turtle_tracker launch_with_action.xml
-->
<launch>
  <!-- Argumentos del launch (posición de explorer) -->
  <arg name="explorer_x" default="2.0" description="Posición X inicial de explorer (0.0-11.0)"/>
  <arg name="explorer_y" default="2.0" description="Posición Y inicial de explorer (0.0-11.0)"/>
  <arg name="explorer_theta" default="0.0" description="Orientación inicial de explorer (radianes)"/>
  
  <!-- Argumentos de control del tracker -->
  <arg name="kp_linear" default="1.5" description="Ganancia proporcional lineal"/>
  <arg name="kp_angular" default="6.0" description="Ganancia proporcional angular"/>
  
  <!-- 1. Iniciar TurtleSim -->
  <node pkg="turtlesim" exec="turtlesim_node" name="turtlesim" output="screen">
  </node>
  
  <!-- 2. Spawner de la tortuga explorer (se ejecuta una vez) -->
  <node pkg="turtle_tracker" exec="turtle_spawner.py" name="turtle_spawner" output="screen">
    <param name="explorer_x" value="$(var explorer_x)"/>
    <param name="explorer_y" value="$(var explorer_y)"/>
    <param name="explorer_theta" value="$(var explorer_theta)"/>
  </node>
  
  <!-- 3. Nodo de seguimiento (tracker) -->
  <node pkg="turtle_tracker" exec="turtle_tracker.py" name="turtle_tracker" output="screen">
    <param name="kp_linear" value="$(var kp_linear)"/>
    <param name="kp_angular" value="$(var kp_angular)"/>
    <param name="max_linear_vel" value="3.0"/>
    <param name="max_angular_vel" value="4.0"/>
    <param name="stop_distance" value="0.3"/>
  </node>
  
  <!-- 4. Action Server de información -->
  <node pkg="turtle_tracker" exec="turtle_info_action_server.py" name="turtle_info_action_server" output="screen">
  </node>
  
  <!-- 5. Action Client (se puede iniciar manualmente para solicitar información) -->
  <!-- Descomentado si se desea que inicie automáticamente -->
  <!--
  <node pkg="turtle_tracker" exec="turtle_info_action_client.py" name="turtle_info_action_client" output="screen">
  </node>
  -->
  
</launch>
